kotlin/internal/jvm/compile.bzl

```diff
@@ -609,7 +609,7 @@ def kt_jvm_produce_jar_actions(ctx, rule_kind):
 
     java_info = JavaInfo(
         output_jar = output_jar,
-        compile_jar = compile_jar,
```

Comment:
This is going to make the compile jar the same as the output jar which will break the whole header / ABI jars for compile avoidance optimization.

Are you expecting any auxiliary files to be included in the compile jars? Note that the final binary (e.g: {java,android}_binary) will link in the output jar for each target, not the compile jar. That is intended only for compilation and why we do this optimization.

A test case showing what is broken would be helpful.

Reply:
Yes, I'm expecting auxiliary files in the output_jar to be available for use by follow-on targets.

A specific example is doing a wire_proto compile which generates the jar containing the generated .class files as well as the .proto files.

A follow-on target uses takes as input this jar and requires both the .proto and .class files as input.

Without this PR, the .proto files get removed when the ijar / hjar file is generated prior to compiling the follow-on target.

An alternate solution would be to include the srcjar as an input to the follow-on target, but the output jar is really the proper generated target.

Reply:
Is there a way to force the output.jar to be used as a dep in a follow-on target build?

Reply:
Is `wire_proto` a Skylark rule?

I'm guessing that `wire_proto` invokes some tool to compile its local proto source into either Kotlin/Java source code or direct to JVM bytecode?

If the compiler needs access to transitive proto source, i.e: local proto source references protos in other libraries then you can make these protos available via a provider e.g: WireProtoInfo that your rule can then collect transitive instances of.